Tests/SourceKitLSPTests/SwiftInterfaceTests.swift

```diff
@@ -135,6 +189,80 @@ final class SwiftInterfaceTests: XCTestCase {
     )
   }
 
+  func testSemanticFunctionalityInGeneratedInterface() async throws {
+    let project = try await SwiftPMTestProject(
+      files: [
+        "MyLibrary/MyLibrary.swift": """
+        public struct Lib {
+          public func foo() -> String {}
+          public init() {}
+        }
+        """,
+        "Exec/main.swift": "import 1️⃣MyLibrary",
+      ],
+      manifest: """
+        let package = Package(
+          name: "MyLibrary",
+          targets: [
+            .target(name: "MyLibrary"),
+            .executableTarget(name: "Exec", dependencies: ["MyLibrary"])
+          ]
+        )
+        """,
+      capabilities: ClientCapabilities(experimental: [
+        "workspace/getReferenceDocument": .bool(true)
+      ]),
+      enableBackgroundIndexing: true
+    )
+
+    let (mainUri, mainPositions) = try project.openDocument("main.swift")
+    let response =
+      try await project.testClient.send(
+        DefinitionRequest(
+          textDocument: TextDocumentIdentifier(mainUri),
+          position: mainPositions["1️⃣"]
+        )
+      )
+    let referenceDocumentUri = try XCTUnwrap(response?.locations?.only).uri
+    let referenceDocument = try await project.testClient.send(GetReferenceDocumentRequest(uri: referenceDocumentUri))
+    let (referenceDocumentPositions, expectedReferenceDocumentContents) = DocumentPositions.extract(
```

Comment:
I wouldn't be super surprised if we add/change imports and/or interface printing at some point. Could we just search for eg. Lib/foo/init + get the position of String from the result rather than assuming it's exactly this interface?

Reply:
I guess you’re right 😉